File: Domain/Product.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Product
    {
         
        public Guid Id { get; set;}
        public string Title {get; set; }
        public string Description { get; set; }
         public string Category { get; set; }
         public string Link { get; set; }
         public string Image { get; set; }
         public string MadeBy {get; set; }
         public int Stock {get; set; }
        public DateTime Date { get; set; }
        public virtual ICollection<UserProduct> UserProducts {get; set;}
        public virtual ICollection<ProductComment> ProductComments {get; set;}
         
    }
}

File: client-app/src/App/Models/product.ts
export interface IProductsEnvelope {
    products: IProduct[];
    productCount: number;
  }

export interface IProduct {
    id: string;
    title: string;
    description: string;
    stock: Number;
    madeBy: string;
    image: string;
    link: string;
    category: string;
    date: Date;
    isHost: boolean;
    isLiked: boolean;
    liked: ILikedProduct[];
    productComments: IComment[];
}

export interface IComment {
    id: string;
    createdAt: Date;
    body: string;
    username: string;
    displayName: string;
    image: string;
  }

export interface IProductFormValues extends Partial<IProduct> {
    time?: Date;
}
export class ProductFormValues implements IProductFormValues {
    id?: string = undefined;
    title: string = '';
    category: string = "";
    description: string = "";
    date?: Date = undefined;
    time?: Date = undefined;

    constructor(init?: IProductFormValues) {
        if (init && init.date){
            init.time = init.date
        }
        Object.assign(this, init);
    } 
}

export interface ILikedProduct {
    username: string;
    displayName: string;
    image: string;
    isHost: boolean;
    following?: boolean;
}